src/constraints.jl

```diff
@@ -1280,3 +1280,51 @@ function num_constraints(model::Model; count_variable_in_set_constraints::Bool)
     end
     return ret
 end
+
+"""
+    all_constraints(model::Model; include_variable_in_set_constraints::Bool)
+
+Return a list of all constraints in `model`.
+
+If `include_variable_in_set_constraints == true`, then `VariableRef` constraints
+such as `VariableRef`-in-`Integer` are included. To return only the structural
+constraints (e.g., the rows in the constraint matrix of a linear program), pass
+`include_variable_in_set_constraints = false`.
```

Comment:
I would add a warning saying that this is not the best approach if performance is a concern

Reply:
I don't think we need to say this. It's unlikely that getting a list of all the constraints will be a performance bottleneck, and we shouldn't be scaring people into more complicated code just for the sake of performance.

Reply:
Why unlikely ? We've had issues with MOI packages not doing this function barrier and getting performance issue

Reply:
Doing it on a critical path within a package is different to a user of JuMP.

Reply:
Users of JuMP might be writing packages as well

Reply:
yes there has been a whole phase of encouraging people to just write libraries on top of JuMP

Reply:
How's this now?